src/payment/payment-action-creator.ts

```diff
@@ -47,21 +60,27 @@ export default class PaymentActionCreator {
         );
     }
 
-    initializeOffsitePayment(
-        methodId: string,
-        gatewayId?: string,
-        instrumentId?: string,
-        shouldSaveInstrument?: boolean,
-        target?: string,
-        promise?: Promise<undefined>
-    ): ThunkAction<InitializeOffsitePaymentAction, InternalCheckoutSelectors> {
+    initializeOffsitePayment: InitializeOffsitePayment = ({
+        methodId,
+        gatewayId,
+        instrumentId,
+        target,
+        promise,
+        shouldSaveInstrument,
+        setAsDefaultInstrument,
+    }) => {
         return store => {
             let paymentData: FormattedPayload<FormattedHostedInstrument | FormattedVaultedInstrument> | undefined;
 
             if (instrumentId) {
                 paymentData = { formattedPayload: { bigpay_token: instrumentId } };
             } else if (shouldSaveInstrument) {
-                paymentData = { formattedPayload: { vault_payment_instrument: shouldSaveInstrument } };
+                paymentData = {
+                    formattedPayload: {
+                        vault_payment_instrument: shouldSaveInstrument,
+                        set_as_default_stored_instrument: setAsDefaultInstrument || null,
```

Comment:
since `setAsDefaultInstrument` only makes sense as part of `shouldSaveInstrument`, what do you think about making `shouldSaveInstrument` an object?


```js
shouldSaveInstrument?: boolean | {
  isDefault: boolean;
}
```

Reply:
`setAsDefaultInstrument` can be with payments instruments that are _already_ vaulted, so it's not necessarily dependant on a `shouldSaveInstrument` value. As such, I'd probably not nest them in each other. We could consider 'grouping' them in a `vaulting` (or similar) namespace? But as there's only two of them I'd probably leave that for later improvement.

On your other point, I'll look as changing it to `shouldSetAsDefaultInstrument`

Reply:
or just `shouldDefaultInstrument` ?

Reply:
Either really, whatever sounds better for you. I was trying to keep the interface consistent as we prefix "commands" with `should`

Reply:
went for the more explicit `shouldSetAsDefaultInstrument` in the end